source/slang/slang-ir.h

```diff
@@ -1856,8 +1856,10 @@ struct IRStructField : IRInst
 struct IRStructType : IRType
 {
     IRFilteredInstList<IRStructField> getFields() { return IRFilteredInstList<IRStructField>(getChildren()); }
-
+    
     IR_LEAF_ISA(StructType)
+
+    IRFunc* defaultCtor = nullptr;
```

Comment:
We shouldn't be adding fields to IR insts like this. All references must be encoded as either a child or an operand.

And IRStruct should not be referencing ctors directly. You can use a decoration to associate a default ctor if necessary. But i don't think we should need this, because we should be generating all default ctor calls during IR lowering, and we should never need to have the concept of a ctor in the IR.

Reply:
I think this is a hole in our generics system that we don't have to address in this PR. 

What we probably should do now is that if T doesn't have a default initializer from any constraints, then we don't call any default ctor even after specialization. However, we still want to make sure we zero initialize whatever type it became after specialization with defaultConstruct inst.


So when we lower the generic function to IR, we check if T has default ctor, if it has we call it. If not we emit a defaultConstruct inst.

Reply:
If we do this, we don't need to keep track of the default ctor of each struct type in the IR in any explicit way.

Reply:
The workaround added was not to handle generics, generics work as intended with __default.

Passing a struct<->Ctor was done because of the cases like `vector<...struct...>`. We don't exactly know what is inside the container, so we cannot simply call an 'invoke' to the ctor. Only __default has emitting logic to manage these code edge cases which is why a ctor was passed through.

Reply:
Well,if you have:
```
struct MyContainer<T>
{
     T obj;
}
```
Then the right answer is always do `obj = __default<T>()`, which will eventually lower into `defaultConstruct`.

If you have:
```
struct MyContainer<T:IDefaultInitializable> // or any other interface that leads to `__init()` to be found
{
    T obj;
}
```
Then you lower it into:
```
obj = T();
```

Reply:
In either case, you shouldn't need to keep track of this additional ctor info.